File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.java
class Solution {
    public int valueAfterKSeconds(int n, int k) {
        long[] a = new long[n];
        for(int i=0;i<n;i++)
            a[i]=1;
        int time =0;
        while(time<k){
            for(int i=1;i<n;i++)
                a[i] = (a[i-1] + a[i])%((int)1e9 + 7);
            time++;
        }
        return (int)(a[n-1]%((int)1e9 + 7));
        
    }
}

File: 3179-find-the-n-th-value-after-k-seconds/3179-find-the-n-th-value-after-k-seconds.cpp
class Solution {
public:
    int md=1e9+7;
    int valueAfterKSeconds(int n, int k) {
       vector<int> arr(n, 1);  
        for (int i=0;i<k;i++) {
            for(int j=1;j<n;j++){
                arr[j]=(arr[j]+arr[j-1])%md;
            }
        }
        
        return arr[n-1];
    }
};